SpatialGDK/Source/SpatialGDKServices/Private/SSpatialOutputLog.cpp

```diff
@@ -254,21 +255,58 @@ void SSpatialOutputLog::StartPollTimer(const FString& LogFilePath)
 	});
 }
 
+void SSpatialOutputLog::FormatAndPrintRawErrorLine(const FString& LogLine)
+{
+	const FRegexPattern ErrorPattern = FRegexPattern(TEXT("level=(.*) msg=(.*) code=(.*) code_string=(.*) error=(.*) stack=(.*)"));
+	FRegexMatcher ErrorMatcher(ErrorPattern, LogLine);
+
+	if (!ErrorMatcher.FindNext())
+	{
+		UE_LOG(LogSpatialOutputLog, Error, TEXT("Failed to parse log line: %s"), *LogLine);
+		return;
+	}
+
+	FString ErrorLevelText = ErrorMatcher.GetCaptureGroup(1);
+	FString Message = ErrorMatcher.GetCaptureGroup(2);
+	FString ErrorCode = ErrorMatcher.GetCaptureGroup(3);
+	FString ErrorCodeString = ErrorMatcher.GetCaptureGroup(4);
+	FString ErrorMessage = ErrorMatcher.GetCaptureGroup(5);
+	FString Stack = ErrorMatcher.GetCaptureGroup(6);
+
+	// The stack message comes with double escaped characters.
+	Stack = Stack.ReplaceEscapedCharWithChar();
+
+	// Format the log message to be easy to read.
+	FString LogMessage = FString::Printf(TEXT("%s \n Code: %s \n Code String: %s \n Error: %s \n Stack: %s"), *Message, *ErrorCode, *ErrorCodeString, *ErrorMessage, *Stack);
+
+	ELogVerbosity::Type LogVerbosity = ELogVerbosity::Error;
+
+	// There are currently no logger names included in the errors from Spatial Service and so we always give it this LogCategory.
+	FString LogCategory = TEXT("SpatialService");
+
+	// Serialization must be done on the game thread.
+	AsyncTask(ENamedThreads::GameThread, [this, LogMessage, LogVerbosity, LogCategory]
+	{
+		Serialize(*LogMessage, LogVerbosity, FName(*LogCategory));
+	});
+}
+
 void SSpatialOutputLog::FormatAndPrintRawLogLine(const FString& LogLine)
 {
 	// Log lines have the format time=LOG_TIME level=LOG_LEVEL logger=LOG_CATEGORY msg=LOG_MESSAGE
-	const FRegexPattern LogPattern = FRegexPattern(TEXT("level=(.*) logger=(.*\\.)?(.*) msg=(.*)"));
+	const FRegexPattern LogPattern = FRegexPattern(TEXT("level=(.*) msg=(.*) loggerName=(.*\\.)?(.*)"));
 	FRegexMatcher LogMatcher(LogPattern, LogLine);
 
 	if (!LogMatcher.FindNext())
 	{
-		UE_LOG(LogSpatialOutputLog, Error, TEXT("Failed to parse log line: %s"), *LogLine);
+		// If this line could not be matched then it's most likely an error with a stack trace. Attempt to use the ErrorLine formatter.
+		FormatAndPrintRawErrorLine(LogLine);
 		return;
 	}
 
 	FString LogLevelText = LogMatcher.GetCaptureGroup(1);
-	FString LogCategory = LogMatcher.GetCaptureGroup(3);
-	FString LogMessage = LogMatcher.GetCaptureGroup(4);
+	FString LogMessage = LogMatcher.GetCaptureGroup(2);
+	FString LogCategory = LogMatcher.GetCaptureGroup(4);
```

Comment:
What happened to 3?

Reply:
They changed `logger` to `loggerName` and changed it's location. The format of this `loggerName` is `ParentName.ChildName.ChildsChildName`, for our purposes we only care about the `ChildsChildName` which is how it's always been. In the regex capture group 3 refers to `ParentName.ChildName` and capture group 4 refers to `ChildsChildName` and there you go